arangod/Cluster/ClusterInfo.cpp

```diff
@@ -4232,6 +4288,83 @@ Result ClusterInfo::finishModifyingAnalyzerCoordinator(
   return Result(TRI_ERROR_NO_ERROR);
 }
 
+void ClusterInfo::initMetricsUpdate() {
+  VPackBuilder builderNewRebootId;
+  builderNewRebootId.add(VPackValue{kInvalidRebootId});
+  VPackBuilder builderNewServerId;
+  builderNewServerId.add(VPackValue{kInvalidServerId});
+
+  AgencyWriteTransaction const write{
+      {{kMetricsUpdateRebootId, AgencyValueOperationType::SET,
+        builderNewRebootId.slice()},
+       {kMetricsUpdateServerId, AgencyValueOperationType::SET,
+        builderNewServerId.slice()}},
+      {{kMetricsUpdateRebootId, AgencyPrecondition::Type::EMPTY, true},
+       {kMetricsUpdateServerId, AgencyPrecondition::Type::EMPTY, true}}};
+  AgencyComm ac{_server};
+  while (true) {
```

Comment:
let's try only several times and also honor server shutdown

Reply:
I think its ok if we can't start coordinator without successful connection to agency

Reply:
But about graceful shutdown good point
Done